cumulus/client/consensus/aura/src/collators/lookahead.rs

```diff
@@ -107,6 +107,9 @@ pub struct Params<BI, CIDP, Client, Backend, RClient, CHP, SO, Proposer, CS> {
 	pub authoring_duration: Duration,
 	/// Whether we should reinitialize the collator config (i.e. we are transitioning to aura).
 	pub reinitialize: bool,
+	/// Whether elastic scaling is enabled for this collation.
+	/// If it is, the collator will send the parent-head data along with the collation.
+	pub with_elastic_scaling: bool,
```

Comment:
Can we please get rid of this `with_elastic_scaling` everywhere in this pr? This is just "dirty". The collation generation logic internally can just decide based on the fact if the `head` is passed or not if it wants to use the new message. 

From all the collator implementations we want to set the `head` to `None` for now and then later do this properly.

Reply:
sure, pushed [94ba49b](https://github.com/paritytech/polkadot-sdk/pull/3302/commits/94ba49bfa00a4f0ce68c2dc77f95364772e84e11)
> The collation generation logic internally can just decide based on the fact if the head is passed or not if it wants to use the new message

could you elaborate on this? passing the head from where?
i mean we could also detect elastic scaling by checking all the assignments for the para_id in collation-generation (or somewhere else)
we could also set to Some unconditionally (which seems wasteful)

Reply:
https://github.com/paritytech/polkadot-sdk/pull/3302/files#diff-14f4394aa7a3666c1a78fb28ccf076505323c497acf8dae788d3fc28a9cc6a18R862 you are already doing here what I meant.

Reply:
And then give `maybe_parent_head` better docs to explain on when to set this.

Reply:
currently, in this PR, `maybe_parent_head` is being set by collation-generation - it's hardcoded to `None`: https://github.com/paritytech/polkadot-sdk/pull/3302/files#diff-d7298fb050b10e14df59080885ebee73861d85b2c4722cc00089fc91a703d63eR543. I've tried to make it configurable with `with_elastic_scaling` param, but you seem to think it's a hack. So my question is, do you have a suggestion on how it should be configured properly?

Reply:
I'll make it part of https://github.com/paritytech/polkadot-sdk/blob/6185b002dfb54c73881257913a830416f2e17a15/polkadot/node/primitives/src/lib.rs#L419 struct produced as part of `CollatorFn` if that the right way.

Reply:
> I'll make `maybe_parent_head_data` part of

Maybe we don't need this, because `CollatorFn` is sort of deprecated I would say. I mean the lookahead collator isn't using this anymore and any feature work will also not use `CollatorFn` anymore. `SubmitCollationParams` is the struct that should be updated to take `maybe_parent_head_data`.

Reply:
Ok, I think I misunderstood you a bit and after chatting with @skunert, settled on approach where collators don't configure anything and elastic scaling is detected automatically (not done in this PR). 
Updated the PR and description. PTAL 🙏